### ISSUE TITLE
Questions about training

### ISSUE THREAD

**Issue by goddoe** [UKPLab/EACL21-personalized-conversational-system#3]
Hi, I read your paper and I have some questions about it.

## Question 1) theoretical background of your actor-critic learning

In your paper, you experimented REINFORCE and actor-critic. As far as I know, naive REINFORCE and actor-critic is on-policy RL method. So, it needs to simulate the environment using action sampled from the policy. But, there is no environment that can simulate dialogue. Also, your paper and your code seem that you train your model without simulation. It seems off-policy learning.

I think that, in this setting, policy can not learn the influence derived from the action sampled from the policy

So, my question is what is the theoretical background of training REINFORCE and actor-critic in this way that doesn't use simulation.

## Question 2) In equation 10, Why do you use the immediate reward(R) only?

To the best of my knowledge, when you use policy gradient using REINFORCE or Actor-critic, you should use accumulated expected return from a state to the end of the scenario which like an accumulated reward from Monte-Carlo sampling or TD learning (r+v(s_{T+1})) - v(s_{T}).

In equation 10, however, you just use immediate reward R, and also your code doesn't contain accumulation or TD.
Is there any theoretical background on it?

I have little knowledge of RL, so I can be wrong. If my question is wrong, please tell me.

**Comment by MMesgar** [comment]
Thanks for your interest to our work.

The theoretical background to answers of your questions is well explained in http://rail.eecs.berkeley.edu/deeprlcourse/ 

Best,

**Comment by goddoe** [comment]
Thanks for the reply and the link!

I checked your link,  I watched [this](https://www.youtube.com/watch?v=wr00ef_TY6Q&list=PL_iWQOsE6TfURIIhCrlt-wj9ByIVpbfGc&index=23) video and also [this](http://rail.eecs.berkeley.edu/deeprlcourse-fa20/static/slides/lec-6.pdf) slide.

In the lecture, when they use actor-critic, they also use 'rewards to go'. My question 2 is just about it. In equation 10 in your paper, your reward function R is the immediate reward. I wonder that why using only immediate R is OK. Is it just your suggestion or Is there any paper using immediate reward or Am I missing something?
